### ISSUE TITLE
Custom Tasks: Move all Run-related Go types to pkg/apis/run

### ISSUE THREAD

**Issue by GregDritschler** [tektoncd/pipeline#3596]
### Feature request

This is a follow-on to #3463.  That PR created a new package `pkg/apis/run/v1alpha1` to solve circular dependency issues between `pkg/apis/pipeline/v1alpha1` and `pkg/apis/pipeline/v1beta1`.  Only RunStatus was moved to the new package because moving the other Run-related types required a fair amount of refactoring to other Tekton types.

### Use case

This affects consumers of the Tekton API who may find it confusing that the Run-related types are split across different packages.

**Comment by GregDritschler** [comment]
I will work on this after #3463 is merged.

**Comment by GregDritschler** [comment]
I investigated this further.  The problem I am running into now is that the tools that generate the client, listers, informers, and reconciler do not support type aliases.  So moving Run completely to a new package means these generated artifacts also change packages with no easy way to keep aliases around in the old package for compatibility.  Since the kubeflow team is already implementing custom task controllers that use these artifacts, I am not in favor of making this breaking change.

**Comment by sbwsg** [comment]
Should we keep this open with a view to consolidating the types in a single package when we move the feature into beta?

**Comment by GregDritschler** [comment]
> Should we keep this open with a view to consolidating the types in a single package when we move the feature into beta?

Yeah I had a similar thought,  though I was thinking in terms of a v1 restructure (probably the next big refactoring).

**Comment by jerop** [comment]
ongoing work 
/lifecycle frozen
